relay-server/src/utils/rate_limits.rs

```diff
@@ -631,19 +640,45 @@ where
             rate_limits.merge(event_limits);
         }
 
+        // Handle attachments.
         if let Some(limit) = enforcement.active_event() {
-            enforcement.attachments =
-                limit.clone_for(DataCategory::Attachment, summary.attachment_quantity);
-        } else if summary.attachment_quantity > 0 {
-            let item_scoping = scoping.item(DataCategory::Attachment);
-            let attachment_limits = self
-                .check
-                .apply(item_scoping, summary.attachment_quantity)?;
-            enforcement.attachments = CategoryLimit::new(
-                DataCategory::Attachment,
-                summary.attachment_quantity,
-                attachment_limits.longest(),
+            let limit1 = limit.clone_for(DataCategory::Attachment, summary.attachment_quantity);
+            let limit2 = limit.clone_for(
+                DataCategory::AttachmentItem,
+                summary.attachment_item_quantity,
             );
+            enforcement.attachments = limit1;
+            enforcement.attachment_items = limit2;
+        } else {
+            let mut attachment_limits = RateLimits::new();
+            if summary.attachment_quantity > 0 {
+                let item_scoping = scoping.item(DataCategory::Attachment);
+
+                let attachment_byte_limits = self
+                    .check
+                    .apply(item_scoping, summary.attachment_quantity)?;
+
+                enforcement.attachments = CategoryLimit::new(
+                    DataCategory::Attachment,
+                    summary.attachment_quantity,
+                    attachment_byte_limits.longest(),
+                );
+                attachment_limits.merge(attachment_byte_limits);
+            }
+            if !attachment_limits.is_limited() && summary.attachment_item_quantity > 0 {
+                let item_scoping = scoping.item(DataCategory::AttachmentItem);
+
+                let attachment_item_limits = self
+                    .check
+                    .apply(item_scoping, summary.attachment_item_quantity)?;
+
+                enforcement.attachment_items = CategoryLimit::new(
+                    DataCategory::AttachmentItem,
+                    summary.attachment_item_quantity,
+                    attachment_item_limits.longest(),
+                );
+                attachment_limits.merge(attachment_item_limits);
+            }
```

Comment:
This function is getting messier with every addition. I will attempt to generalize it in a future PR.